src/Utilities/ProjectData.h

```diff
@@ -23,118 +23,133 @@
+   * translate to project_main_ = "det_qmc_short_sdbatch_vmcbatch_mwalkers.s000"
    */
-  inline const std::string& CurrentMainRoot() const { return m_projectmain; }
+  const std::string& currentMainRoot() const noexcept;
 
-  /** returns the nextroot of the project, the series id is incremented at every QMC section
+  /** 
+   * @brief returns the nextroot of the project, the series id is incremented at every QMC section
    * <project id="det_qmc_short_sdbatch_vmcbatch_mwalkers" series="0">
-   * translate to m_projectmain = "det_qmc_short_sdbatch_vmcbatch_mwalkers.s001"
+   * translate to project_main_ = "det_qmc_short_sdbatch_vmcbatch_mwalkers.s001"
    */
-  inline const std::string& NextRoot() const { return m_nextroot; }
+  const std::string& nextRoot() const noexcept;
 
-  /** return the root of the previous sequence
-   * @param oldroot is composed by the m_title and m_series
+  /** 
+   * @brief return the root of the previous sequence
+   * @param oldroot is composed by the m_title_ and series_index_
    */
-  bool PreviousRoot(std::string& oldroot) const;
+  bool previousRoot(std::string& oldroot) const;
 
-  int getSeriesIndex() const { return m_series; }
-  int getMaxCPUSeconds() const { return max_cpu_secs_; }
-  DriverVersion get_driver_version() const { return driver_version_; }
+  int getSeriesIndex() const noexcept;
+
+  int getMaxCPUSeconds() const noexcept;
+
+  DriverVersion getDriverVersion() const noexcept;
+
+  /**
+   * @brief Check if simulation run is real or complex
+   * true: complex run
+   * false: real run (default)
+   */
+  static bool isComplex() noexcept;
 
 private:
-  static DriverVersion lookupDriverVersion(const std::string& enum_value);
+  static const std::unordered_map<std::string, DriverVersion> lookup_input_enum_value_;
 
   ///title of the project
-  std::string m_title;
+  std::string title_;
 
   ///name of the host where the job is running
-  std::string m_host;
+  std::string host_;
 
   ///date when the job is executed
-  std::string m_date;
+  std::string date_;
 
   ///main root for all the output engines
-  std::string m_projectmain;
+  std::string project_main_;
 
   ///processor-dependent root for all the output engines
-  std::string m_projectroot;
+  std::string project_root_;
 
   ///root for the next run
-  std::string m_nextroot;
+  std::string next_root_;
 
   ///series index
-  int m_series;
+  int series_index_;
 
   ///communicator
-  Communicate* myComm;
+  Communicate* my_comm_;
 
   ///the xml node for <Project/>
-  xmlNodePtr m_cur;
+  xmlNodePtr m_cur_;
 
   ///max cpu seconds
   int max_cpu_secs_;
 
   // The driver version of the project
   DriverVersion driver_version_;
+
+  /**
+   * @brief runtime variable to track real or complex runs using a single compiled qmcpack executable
+   * @todo remove/rethink its "const" nature
+   * true: complex run
+   * false: real run (default)
+   */
+  static const bool is_complex_;
```

Comment:
This shouldn't be static.  Especially in testing I should be able to construct both a real and a complex ProjectData depending on the input.

Reply:
Agreed.